fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/PartitionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,159 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.table.Column
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import com.google.common.collect.Sets
+
+import org.joda.time.Interval
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.util.function.Function
+
+
+/**
+ * Test for partition availability behavior.
+ */
+class PartitionAvailabilitySpec extends Specification{
+
+    PartitionAvailability partitionAvailability
+
+    Availability availability1
+    Availability availability2
+
+    Column column1
+    Column column2
+
+    def setup() {
+        availability1 = Mock(Availability)
+        availability2 = Mock(Availability)
+    }
+
+    @Unroll
+    def "getDataSourceNames returns #expected from #dataSourceNames1 and #dataSourceNames2"() {
+        given:
+        availability1.getDataSourceNames() >> Sets.newHashSet(dataSourceNames1.collect{it -> TableName.of(it)})
+        availability2.getDataSourceNames() >> Sets.newHashSet(dataSourceNames2.collect{it -> TableName.of(it)})
+
+        partitionAvailability = new PartitionAvailability(
+                [availability1, availability2] as Set,
+                Collections.emptySet(),
+                {}
+        )
+
+        expect:
+        partitionAvailability.getDataSourceNames() == Sets.newHashSet(expected.collect{it -> TableName.of(it)})
+
+        where:
+        dataSourceNames1       | dataSourceNames2 | expected
+        []                     | []               | []
+        ['source1']            | []               | ['source1']
+        ['source1']            | ['source1']      | ['source1']
+        ['source1', 'source2'] | ['source1']      | ['source1', 'source2']
+    }
+
+    @Unroll
+    def "getAllAvailableIntervals returns merged intervals grouped by columns when two availabilities have #caseDescription"() {
+        given:
+        column1 = new Column('col1')
+        column2 = new Column(column2Name)
+
+        availability1.getAllAvailableIntervals() >> [
+                (column1): intervallist1.collect{it -> new Interval(it)}
+        ]
+        availability2.getAllAvailableIntervals() >> [
+                (column2): intervallist2.collect{it -> new Interval(it)}
```

Comment:
No need to wrap this if you don't have the necessary `it -> ` in the closures: `[(column2): intervallist2.collect {new Interval(it)}]`